biz.aQute.bndlib/src/aQute/bnd/classfile/ClassFile.java

```diff
@@ -153,4 +166,28 @@ static ByteBuffer slice(DataInput in, int length) throws IOException {
 		in.readFully(array, 0, length);
 		return ByteBuffer.wrap(array, 0, length);
 	}
+
+	public byte[] toBytes() throws IOException {
+		ByteArrayOutputStream bout = new ByteArrayOutputStream();
+		DataOutputStream dout = new DataOutputStream(bout);
+		write(dout);
+		return bout.toByteArray();
+	}
+
+	public static ClassFile parseFile(File file) throws FileNotFoundException, IOException {
+		return parseInputStream(new FileInputStream(file));
+	}
+
+	/**
+	 * Rename classes referenced in this class file by their binary name. This
+	 * will change all attributes, and references that refer to class names. The
+	 * mapper must also be useful for the optional `.class` suffix used for
+	 * class paths.
+	 *
+	 * @param mapper maps a binary class name to either null or a different name
+	 */
+	public Optional<ClassFile> rename(Function<String, String> mapper) {
```

Comment:
The classfile package is carefully structure for read-only operations. All mutation operation are kept outside of this package. I would prefer to continue this. So we should move all mutation code to a subpackage such as aQute.bnd.classfile.builder or some other subpackage. I want to be clear that the classfile package is not about mutation.

Reply:
not sure I agree ... but ok